apps/frontend/src/pages/admin/affiliates.vue

```diff
@@ -0,0 +1,281 @@
+						@revoke="revokeAffiliateCode"
+					/>
+				</div>
+			</Accordion>
+		</div>
+	</div>
+</template>
+<script setup lang="ts">
+import { PlusIcon, SearchIcon, XCircleIcon, XIcon } from '@modrinth/assets'
+import {
+	Accordion,
+	Admonition,
+	AffiliateLinkCard,
+	AffiliateLinkCreateModal,
+	Avatar,
+	Button,
+	ButtonStyled,
+	ConfirmModal,
+	injectNotificationManager,
+} from '@modrinth/ui'
+import type { AffiliateLink, User } from '@modrinth/utils'
+
+const { handleError } = injectNotificationManager()
+
+type UserGroup = {
+	user: User
+	affiliates: AffiliateLink[]
+}
+
+const createModal = useTemplateRef<typeof AffiliateLinkCreateModal>('createModal')
+const revokeModal = useTemplateRef<typeof ConfirmModal>('revokeModal')
+
+const {
+	data: affiliateCodes,
+	error,
+	refresh,
+} = await useAsyncData(
+	'AffiliateLinks',
+	() => useBaseFetch('affiliate', { method: 'GET', internal: true }) as Promise<AffiliateLink[]>,
+)
+
+const filterQuery = ref('')
+const creatingLink = ref(false)
+
+const userIds = computed(() => {
+	if (!affiliateCodes.value) {
+		return []
+	}
+	const ids = new Set<string>()
+	affiliateCodes.value.forEach((code) => {
+		ids.add(code.affiliate)
+		ids.add(code.created_by)
+	})
+	return Array.from(ids)
+})
+
+const { data: users } = await useAsyncData(
+	'admin-affiliates-bulk-users',
+	() => {
+		if (userIds.value.length === 0) return Promise.resolve([])
+		return useBaseFetch(`users?ids=${JSON.stringify(userIds.value)}`) as Promise<User[]>
+	},
+	{
+		watch: [userIds],
+	},
+)
+
+const userMap = computed(() => {
+	if (!users.value) {
+		return new Map()
+	}
+	return new Map(users.value.map((user) => [user.id, user]))
+})
+
+const groupedAffiliates = computed((): UserGroup[] => {
+	if (!affiliateCodes.value || !users.value) {
+		return []
+	}
+
+	const groups = new Map<string, UserGroup>()
+
+	affiliateCodes.value.forEach((code) => {
+		const user = userMap.value.get(code.affiliate)
+		if (!user) {
+			return
+		}
+
+		if (!groups.has(user.id)) {
+			groups.set(user.id, {
+				user,
+				affiliates: [],
+			})
+		}
+
+		groups.get(user.id)!.affiliates.push(code)
+	})
+
+	return Array.from(groups.values()).sort((a, b) => a.user.username.localeCompare(b.user.username))
+})
+
+const filteredGroupedAffiliates = computed(() => {
+	if (!filterQuery.value.trim()) {
+		return groupedAffiliates.value
+	}
+
+	const query = filterQuery.value.trim().toLowerCase()
+	return groupedAffiliates.value.filter(
+		(group) =>
+			group.user.username.toLowerCase().includes(query) ||
+			group.affiliates.some((affiliate) => affiliate.source_name.toLowerCase().includes(query)),
+	)
+})
+
+function getCreatedByUsername(createdBy: string): string {
+	const user = userMap.value.get(createdBy)
+	return user?.username || 'Unknown'
+}
+
+async function createAffiliateCode(data: { sourceName: string; username?: string }) {
+	creatingLink.value = true
+
+	try {
+		if (!data.username) {
+			// noinspection ExceptionCaughtLocallyJS
```

Comment:
fine for now, but you should configure your IDE to use eslint rather than jetbrains linting